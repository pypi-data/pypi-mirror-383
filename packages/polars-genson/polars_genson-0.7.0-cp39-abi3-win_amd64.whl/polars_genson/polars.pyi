"""Extend polars DataFrame with genson namespace."""

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    import polars as pl
    from . import GensonNamespace

    # This extends DataFrame with the genson attribute
    class DataFrameNamespaceExtension:
        genson: "GensonNamespace"

    # Monkey patch DataFrame for type checking
    class DataFrame(pl.DataFrame, DataFrameNamespaceExtension):
        pass

    # Replace DataFrame in polars module
    pl.DataFrame = DataFrame  # type: ignore[assignment]
