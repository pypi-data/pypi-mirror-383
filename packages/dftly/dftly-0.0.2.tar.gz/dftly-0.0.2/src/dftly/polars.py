"""Polars execution engine for dftly."""

from __future__ import annotations

from typing import Any, Dict, Mapping
from datetime import datetime, timezone

try:
    import polars as pl
except ModuleNotFoundError as exc:  # pragma: no cover - import guard
    raise ModuleNotFoundError(
        "Polars is required for the dftly.polars module. Install with 'dftly[polars]'"
    ) from exc

from .expressions import ExpressionRegistry
from .nodes import Column, Expression, Literal


_CLOCK_TIME = object()


_TYPE_MAP: dict[str, Any] = {
    "int": int,
    "integer": int,
    "float": float,
    "double": float,
    "bool": bool,
    "boolean": bool,
    "str": str,
    "string": str,
    "date": pl.Date,
    "datetime": pl.Datetime,
    "duration": pl.Duration,
    "time": _CLOCK_TIME,
    "clock_time": _CLOCK_TIME,
    "clock-time": _CLOCK_TIME,
}


def to_polars(node: Any) -> pl.Expr:
    """Convert a dftly node to a polars expression."""
    if isinstance(node, Literal):
        return pl.lit(node.value)
    if isinstance(node, Column):
        return pl.col(node.name)
    if isinstance(node, Expression):
        return _expr_to_polars(node)
    raise TypeError(f"Unsupported node type: {type(node).__name__}")


def map_to_polars(mapping: Mapping[str, Any]) -> Dict[str, pl.Expr]:
    """Convert a mapping of dftly nodes to polars expressions."""
    return {k: to_polars(v) for k, v in mapping.items()}


def _ensure_naive_datetime(dt: datetime) -> datetime:
    """Return a timezone-naive datetime in UTC for subtraction or casting."""

    if dt.tzinfo is None:
        return dt
    return dt.astimezone(timezone.utc).replace(tzinfo=None)


def _expr_to_polars(expr: Expression) -> pl.Expr:
    return ExpressionRegistry.to_polars(expr, to_polars, pl=pl)


def _resolve_timestamp(args: Mapping[str, Any]) -> pl.Expr:
    date = args["date"]
    time = args["time"]

    if isinstance(date, Mapping):
        year = to_polars(date["year"])
        month = to_polars(date["month"])
        day = to_polars(date["day"])
    else:
        date_expr = to_polars(date)
        year = date_expr.dt.year()
        month = date_expr.dt.month()
        day = date_expr.dt.day()

    base_datetime = pl.datetime(year, month, day, 0, 0, 0)

    if isinstance(time, Mapping):

        def _component(name: str) -> pl.Expr:
            node = time.get(name)
            return to_polars(node) if node is not None else pl.lit(0)

        hour = _component("hour")
        minute = _component("minute")
        second = _component("second")
        return pl.datetime(year, month, day, hour, minute, second)

    return base_datetime + to_polars(time)


__all__ = ["to_polars", "map_to_polars"]
