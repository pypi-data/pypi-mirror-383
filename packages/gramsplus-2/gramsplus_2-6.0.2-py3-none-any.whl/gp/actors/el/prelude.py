from gp.actors.el.cangen import CanGenActor, CanGenActorArgs
from gp.actors.el.canrank import CanRankActor, CanRankActorArgs
from gp.actors.el.canreg import CanRegActor, CanRegActorArgs

__all__ = [
    "CanGenActor",
    "CanGenActorArgs",
    "CanRankActor",
    "CanRankActorArgs",
    "CanRegActor",
    "CanRegActorArgs",
]
