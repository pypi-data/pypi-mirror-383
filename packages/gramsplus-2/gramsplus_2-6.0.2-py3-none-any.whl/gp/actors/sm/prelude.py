from gp.actors.sm.can_graph import CanGraphActor, CanGraphActorArgs

__all__ = [
    "CanGraphActor",
    "CanGraphActorArgs",
]
