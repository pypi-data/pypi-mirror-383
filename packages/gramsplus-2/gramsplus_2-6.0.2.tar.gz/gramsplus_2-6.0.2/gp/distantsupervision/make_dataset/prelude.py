from .erv1 import EntityRecognitionV1, EntityRecognitionV1Args
from .fbhct import FilterByHeaderColType, FilterByHeaderColTypeArgs
from .fbt import FilterByEntType, FilterByEntTypeArgs
from .fcom import CombinedFilter, CombinedFilterArgs
from .fentcol import FilterNotEntCol
from .freg import FilterRegex, FilterRegexArgs
from .fv1 import FilterV1, FilterV1Args
from .interface import FilterFn, LabelFn, NoFilter, NoTransform, TransformFn
from .lv1 import LabelV1, LabelV1Args
from .lv2 import LabelV2, LabelV2Args
from .mv1 import TransformV1, TransformV1Args
from .mv2 import TransformV2
from .utils import FilterMixin

__all__ = [
    "FilterFn",
    "LabelFn",
    "NoFilter",
    "FilterMixin",
    "FilterRegex",
    "FilterRegexArgs",
    "FilterV1",
    "FilterV1Args",
    "FilterByEntType",
    "FilterByEntTypeArgs",
    "FilterByHeaderColType",
    "FilterByHeaderColTypeArgs",
    "FilterNotEntCol",
    "CombinedFilter",
    "CombinedFilterArgs",
    "TransformFn",
    "NoTransform",
    "TransformV1",
    "TransformV1Args",
    "TransformV2",
    "LabelV1",
    "LabelV1Args",
    "LabelV2",
    "LabelV2Args",
    "EntityRecognitionV1",
    "EntityRecognitionV1Args",
]
