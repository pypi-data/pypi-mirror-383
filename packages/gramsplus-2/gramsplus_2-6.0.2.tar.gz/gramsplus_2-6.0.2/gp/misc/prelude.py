from gp.misc.appconfig import AppConfig

__all__ = ["AppConfig"]
