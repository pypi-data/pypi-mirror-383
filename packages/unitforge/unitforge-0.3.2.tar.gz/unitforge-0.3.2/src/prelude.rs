use ndarray::{Array1, Array2, ArrayView1, ArrayView2};
use num_traits::{FromPrimitive, Zero};
use std::ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Sub, SubAssign};

#[macro_export]
macro_rules! power {
    ($x:expr, 1) => {
        $x
    };
    ($x:expr, 2) => {
        $x * $x
    };
    ($x:expr, 3) => {
        $x * $x * $x
    };
    ($x:expr, 4) => {
        $x * $x * $x * $x
    };
    ($x:expr, 5) => {
        $x * $x * $x * $x * $x
    };
    ($x:expr, 6) => {
        $x * $x * $x * $x * $x * $x
    };
}

pub trait MulArray1<RHS> {
    type Output;

    fn mul_array1(self, rhs: Array1<RHS>) -> Self::Output;
}

pub trait MulArray2<RHS> {
    type Output;

    fn mul_array2(self, rhs: Array2<RHS>) -> Result<Self::Output, String>;
}

pub trait DivArray1<RHS> {
    type Output;

    fn div_array1(self, rhs: Array1<RHS>) -> Self::Output;
}

pub trait DivArray2<RHS> {
    type Output;

    fn div_array2(self, rhs: Array2<RHS>) -> Result<Self::Output, String>;
}

pub trait QuantityArray2<T> {
    fn from_raw(raw: ArrayView2<f64>, unit: T) -> Self;
    fn to_raw(&self) -> Array2<f64>;
    fn to(&self, unit: T) -> Array2<f64>;
}

pub trait QuantityArray1<T> {
    fn from_raw(raw: ArrayView1<f64>, unit: T) -> Self;
    fn to_raw(&self) -> Array1<f64>;
    fn to(&self, unit: T) -> Array1<f64>;
}

pub trait PhysicsQuantity:
    Copy
    + FromPrimitive
    + Zero
    + Add<Output = Self>
    + AddAssign
    + Div<f64, Output = Self>
    + DivAssign<f64>
    + Mul<f64, Output = Self>
    + MulAssign<f64>
    + Sub<Output = Self>
    + SubAssign
    + PartialOrd
    + Neg<Output = Self>
    + From<f64>
{
    fn as_f64(&self) -> f64;
    type Unit: PhysicsUnit;
    fn new(value: f64, unit: Self::Unit) -> Self;
    fn to(&self, unit: Self::Unit) -> f64;
    fn get_tuple(&self) -> (f64, i32);
    fn abs(self) -> Self;
    fn from_raw(value: f64) -> Self;
    fn from_exponential(multiplier: f64, power: i32) -> Self;
    fn min(self, other: Self) -> Self;
    fn max(self, other: Self) -> Self;
    #[deprecated(since = "0.2.9", note = "please use `as_f64()` instead")]
    fn to_raw(&self) -> f64 {
        self.as_f64()
    }
    #[deprecated(since = "0.2.9", note = "please use `as_f64()` instead")]
    fn get_value(&self) -> f64 {
        self.as_f64()
    }
    fn get_power(&self) -> i32;
    fn get_multiplier(&self) -> f64;
    fn split_value(v: f64) -> (f64, i32);
    fn is_close(&self, other: &Self, tolerance: &Self) -> bool;
    fn optimize(&mut self);
    fn is_nan(&self) -> bool;
    const INFINITY: Self;
    const NEG_INFINITY: Self;
}

pub trait Sqrt<T> {
    fn sqrt(self) -> T;
}

impl PhysicsQuantity for f64 {
    fn as_f64(&self) -> f64 {
        *self
    }

    type Unit = NoUnit;

    fn new(value: f64, _unit: Self::Unit) -> Self {
        value
    }

    fn to(&self, _unit: Self::Unit) -> f64 {
        self.as_f64()
    }

    fn get_tuple(&self) -> (f64, i32) {
        (self.get_multiplier(), self.get_power())
    }

    fn abs(self) -> Self {
        self.abs()
    }

    fn is_nan(&self) -> bool {
        f64::is_nan(*self)
    }

    fn from_raw(value: f64) -> Self {
        value
    }

    fn from_exponential(multiplier: f64, power: i32) -> Self {
        multiplier * 10_f64.powi(power)
    }

    fn min(self, other: Self) -> Self {
        if self < other {
            self
        } else {
            other
        }
    }

    fn max(self, other: Self) -> Self {
        if self > other {
            self
        } else {
            other
        }
    }
    fn get_power(&self) -> i32 {
        0
    }

    fn get_multiplier(&self) -> f64 {
        *self
    }

    fn split_value(v: f64) -> (f64, i32) {
        let power = v.abs().log10().floor() as i32;
        let multiplier = v / 10f64.powi(power);
        (multiplier, power)
    }

    fn is_close(&self, other: &Self, tolerance: &Self) -> bool {
        (self - other).abs() < (*tolerance)
    }

    fn optimize(&mut self) {}

    const INFINITY: Self = f64::INFINITY;
    const NEG_INFINITY: Self = f64::NEG_INFINITY;
}
pub use crate::power;

pub trait PhysicsUnit {
    fn name(&self) -> &str;
    fn base_per_x(&self) -> (f64, i32);
}

pub enum NoUnit {
    no_unit,
}

impl PhysicsUnit for NoUnit {
    fn name(&self) -> &str {
        ""
    }

    fn base_per_x(&self) -> (f64, i32) {
        (1., 0)
    }
}
