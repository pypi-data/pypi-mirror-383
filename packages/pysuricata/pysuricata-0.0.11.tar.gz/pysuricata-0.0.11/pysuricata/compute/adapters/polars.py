"""Polars-specific adapter implementation.

This module provides the polars-specific implementation of the DataAdapter
protocol, optimized for polars DataFrames and Series.
"""

from __future__ import annotations

import logging
from typing import Any, Dict, Optional

import numpy as np

from ..core.types import ColumnKinds
from ..processing.conversion import UnifiedConverter
from ..processing.inference import UnifiedTypeInferrer
from .base import BaseAdapter

try:
    import polars as pl
except ImportError:
    pl = None


class PolarsAdapter(BaseAdapter):
    """Polars-specific adapter implementation.

    This adapter provides optimized operations for polars DataFrames and Series,
    leveraging polars-specific optimizations and features.
    """

    def __init__(
        self,
        converter: Optional[UnifiedConverter] = None,
        inferrer: Optional[UnifiedTypeInferrer] = None,
        logger: Optional[logging.Logger] = None,
    ):
        """Initialize the polars adapter.

        Args:
            converter: Unified data converter.
            inferrer: Unified type inferrer.
            logger: Logger for adapter operations.
        """
        super().__init__(converter, inferrer, logger)

        if pl is None:
            raise ImportError("polars is required for PolarsAdapter")

    def infer_and_build(
        self, data: Any, config: Any
    ) -> tuple[ColumnKinds, Dict[str, Any]]:
        """Infer column types and build accumulators for polars data.

        Args:
            data: Polars DataFrame to analyze.
            config: Configuration object with processing parameters.

        Returns:
            Tuple of (column_kinds, accumulators_dict).
        """
        if not isinstance(data, pl.DataFrame):
            raise TypeError(f"Expected polars.DataFrame, got {type(data)}")

        # Infer column kinds
        result = self.inferrer.infer_kinds(data)
        if not result.success:
            raise RuntimeError(f"Type inference failed: {result.error}")

        kinds = result.data

        # Build accumulators
        accs = {}

        # Import accumulator classes
        from ...accumulators import (
            BooleanAccumulator,
            CategoricalAccumulator,
            DatetimeAccumulator,
            NumericAccumulator,
        )
        from ...accumulators.factory import build_accumulators

        # Build accumulators using the factory
        accs = build_accumulators(kinds, config)

        # Set actual dtypes from the DataFrame
        for col_name in data.columns:
            if col_name in accs:
                actual_dtype = str(data[col_name].dtype)
                accs[col_name].set_dtype(actual_dtype)

        # Apply reclassification logic for numeric columns
        from ..processing.inference import (
            should_reclassify_numeric_as_boolean,
            should_reclassify_numeric_as_categorical,
        )

        # First, apply forced column types to all columns
        if config and config.force_column_types:
            for col_name, forced_type in config.force_column_types.items():
                if col_name in data.columns:
                    if self.logger:
                        self.logger.info(
                            "ðŸ”§ forcing column '%s' type to '%s' (override) [pl]",
                            col_name,
                            forced_type,
                        )

                    # Remove from current category
                    if col_name in kinds.numeric:
                        kinds.numeric.remove(col_name)
                    elif col_name in kinds.categorical:
                        kinds.categorical.remove(col_name)
                    elif col_name in kinds.boolean:
                        kinds.boolean.remove(col_name)
                    elif col_name in kinds.datetime:
                        kinds.datetime.remove(col_name)

                    # Add to forced category
                    if forced_type == "boolean":
                        kinds.boolean.append(col_name)
                        from ...accumulators import BooleanAccumulator

                        accs[col_name] = BooleanAccumulator(col_name)
                    elif forced_type == "categorical":
                        kinds.categorical.append(col_name)
                        from ...accumulators import CategoricalAccumulator

                        accs[col_name] = CategoricalAccumulator(col_name)
                    elif forced_type == "numeric":
                        kinds.numeric.append(col_name)
                        from ...accumulators import NumericAccumulator

                        accs[col_name] = NumericAccumulator(col_name)
                    elif forced_type == "datetime":
                        kinds.datetime.append(col_name)
                        from ...accumulators import DatetimeAccumulator

                        accs[col_name] = DatetimeAccumulator(col_name)

                    accs[col_name].set_dtype(str(data[col_name].dtype))

        for col_name in data.columns:
            if col_name in kinds.numeric:
                # Check if this numeric column should be reclassified as boolean
                if config and should_reclassify_numeric_as_boolean(
                    data[col_name], config, self.logger
                ):
                    if self.logger:
                        self.logger.info(
                            "ðŸ”„ reclassifying '%s' from numeric to boolean (0/1 values detected) [pl]",
                            col_name,
                        )
                    # Move from numeric to boolean
                    kinds.numeric.remove(col_name)
                    kinds.boolean.append(col_name)

                    # Replace accumulator
                    from ...accumulators import BooleanAccumulator

                    accs[col_name] = BooleanAccumulator(col_name)
                    accs[col_name].set_dtype(str(data[col_name].dtype))
                else:
                    # Check if this numeric column should be reclassified as categorical
                    try:
                        unique_count = data[col_name].n_unique()
                        total_count = data.height
                        if should_reclassify_numeric_as_categorical(
                            unique_count, total_count
                        ):
                            if self.logger:
                                self.logger.info(
                                    "ðŸ”„ reclassifying '%s' from numeric to categorical (unique: %d, total: %d) [pl]",
                                    col_name,
                                    unique_count,
                                    total_count,
                                )
                            # Move from numeric to categorical
                            kinds.numeric.remove(col_name)
                            kinds.categorical.append(col_name)

                            # Replace accumulator
                            from ...accumulators import CategoricalAccumulator

                            accs[col_name] = CategoricalAccumulator(col_name)
                            accs[col_name].set_dtype(str(data[col_name].dtype))
                    except Exception as e:
                        if self.logger:
                            self.logger.debug(
                                "Could not check unique values for '%s': %s [pl]",
                                col_name,
                                e,
                            )

        return kinds, accs

    def estimate_mem(self, frame: Any) -> int:
        """Estimate memory usage of a polars DataFrame.

        Args:
            frame: Polars DataFrame to estimate memory for.

        Returns:
            Estimated memory usage in bytes.
        """
        if not isinstance(frame, pl.DataFrame):
            return 0

        try:
            # Use polars estimated_size for accurate estimation
            return int(frame.estimated_size())
        except Exception:
            # Fallback estimation
            return frame.height * len(frame.columns) * 8  # Rough estimate

    def missing_cells(self, frame: Any) -> int:
        """Count missing cells in a polars DataFrame.

        Args:
            frame: Polars DataFrame to count missing cells in.

        Returns:
            Number of missing cells.
        """
        if not isinstance(frame, pl.DataFrame):
            return 0

        try:
            # Optimized missing value counting - use sum_horizontal() instead of complex aggregation
            return int(frame.null_count().sum_horizontal().item())
        except Exception:
            try:
                # Fallback to the original method if sum_horizontal() is not available
                return int(frame.null_count().select(pl.sum(pl.all())).to_numpy()[0][0])
            except Exception:
                return 0

    def consume_chunk(
        self,
        data: Any,
        accs: Dict[str, Any],
        kinds: ColumnKinds,
        config: Optional[Any] = None,
        logger: Optional[Any] = None,
    ) -> None:
        """Consume a polars DataFrame chunk and update accumulators.

        Args:
            data: Polars DataFrame chunk to process.
            accs: Dictionary of accumulators to update.
            kinds: Column type information.
            config: Configuration object for type inference and processing.
            logger: Optional logger for progress tracking.
        """
        if not isinstance(data, pl.DataFrame):
            raise TypeError(f"Expected polars.DataFrame, got {type(data)}")

        # Import the polars-specific consume function
        from ..consume_polars import consume_chunk_polars

        # Use the existing polars consume function
        consume_chunk_polars(data, accs, kinds, config, logger)

    def update_corr(
        self, frame: Any, corr_est: Any, logger: Optional[Any] = None
    ) -> None:
        """Update correlation estimator with polars DataFrame.

        Args:
            frame: Polars DataFrame to process.
            corr_est: Correlation estimator to update.
            logger: Optional logger for progress tracking.
        """
        if not isinstance(frame, pl.DataFrame):
            return

        try:
            if hasattr(corr_est, "update_from_polars"):
                corr_est.update_from_polars(frame)
        except Exception as e:
            if logger:
                logger.warning(f"Correlation update failed: {e}")

    def sample_section_html(self, first: Any, cfg: Any) -> str:
        """Generate HTML for sample data section from polars DataFrame.

        Args:
            first: First polars DataFrame chunk for sampling.
            cfg: Configuration object.

        Returns:
            HTML string for sample section.
        """
        if not isinstance(first, pl.DataFrame):
            return ""

        try:
            # Import the polars-specific render function
            from ...render.sections import render_sample_section_polars

            # Extract sample_rows from config, default to 10
            sample_rows = getattr(cfg, "sample_rows", 10)
            return render_sample_section_polars(first, sample_rows)
        except Exception:
            return ""

    def _is_compatible_data(self, data: Any) -> bool:
        """Check if data is compatible with polars adapter.

        Args:
            data: Data to check.

        Returns:
            True if compatible, False otherwise.
        """
        return isinstance(data, pl.DataFrame)

    def get_polars_info(self) -> Dict[str, Any]:
        """Get polars-specific information.

        Returns:
            Dictionary with polars-specific information.
        """
        info = self.get_adapter_info()
        info.update(
            {
                "polars_version": pl.__version__ if pl else None,
                "numpy_version": np.__version__,
            }
        )
        return info

    def update_row_kmv(self, frame: Any, row_kmv: Any) -> None:
        """Update the row-level distinct estimator from a polars DataFrame.

        Args:
            frame: Polars DataFrame to process.
            row_kmv: Row KMV estimator to update.
        """
        if not isinstance(frame, pl.DataFrame):
            return

        try:
            if hasattr(row_kmv, "update_from_polars"):
                row_kmv.update_from_polars(frame)
        except Exception as e:
            if self.logger:
                self.logger.warning(f"Failed to update row KMV: {e}")
