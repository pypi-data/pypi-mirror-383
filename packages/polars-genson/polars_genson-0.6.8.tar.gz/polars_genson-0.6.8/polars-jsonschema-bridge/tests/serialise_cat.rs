use insta::assert_yaml_snapshot;
use polars::prelude::*;
use polars_jsonschema_bridge::{polars_schema_to_json_schema, JsonSchemaOptions};

#[test]
fn test_categorical_physical_types() {
    let categories_u8 = Categories::new(
        PlSmallStr::from_static("test_u8"),
        PlSmallStr::from_static("test_namespace"),
        CategoricalPhysical::U8,
    );

    let categories_u16 = Categories::new(
        PlSmallStr::from_static("test_u16"),
        PlSmallStr::from_static("test_namespace"),
        CategoricalPhysical::U16,
    );

    let categories_u32 = Categories::new(
        PlSmallStr::from_static("test_u32"),
        PlSmallStr::from_static("test_namespace"),
        CategoricalPhysical::U32,
    );

    let schema = Schema::from_iter(vec![
        Field::new(
            "cat_u8".into(),
            DataType::Categorical(categories_u8, Arc::new(CategoricalMapping::new(255))),
        ),
        Field::new(
            "cat_u16".into(),
            DataType::Categorical(categories_u16, Arc::new(CategoricalMapping::new(65535))),
        ),
        Field::new(
            "cat_u32".into(),
            DataType::Categorical(categories_u32, Arc::new(CategoricalMapping::new(100000))),
        ),
    ]);

    let options = JsonSchemaOptions::new();
    let json_schema = polars_schema_to_json_schema(&schema, &options).unwrap();

    assert_yaml_snapshot!("categorical_physical_types", json_schema);
}
