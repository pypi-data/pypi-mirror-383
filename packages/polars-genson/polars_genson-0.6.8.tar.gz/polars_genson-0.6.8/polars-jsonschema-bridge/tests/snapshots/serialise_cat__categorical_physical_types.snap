---
source: polars-jsonschema-bridge/tests/serialise_cat.rs
expression: json_schema
---
$schema: "https://json-schema.org/draft/2020-12/schema"
type: object
properties:
  cat_u8:
    type: string
    description: Categorical data
  cat_u16:
    type: string
    description: Categorical data
  cat_u32:
    type: string
    description: Categorical data
required:
  - cat_u16
  - cat_u32
  - cat_u8
additionalProperties: false
